sorter.go

```diff
@@ -63,6 +65,9 @@ func (l keyList) Less(i, j int) bool {
 		if ai != bi {
 			return ai < bi
 		}
+		if ar[i] == br[i] {
+			continue
```

Comment:
Can i offer to simple do a string comparison on the last case:
So the less function would look like this:
```
func (l keyList) Less(i, j int) bool {
	a := l[i]
	b := l[j]
	ak := a.Kind()
	bk := b.Kind()
	for (ak == reflect.Interface || ak == reflect.Ptr) && !a.IsNil() {
		a = a.Elem()
		ak = a.Kind()
	}
	for (bk == reflect.Interface || bk == reflect.Ptr) && !b.IsNil() {
		b = b.Elem()
		bk = b.Kind()
	}
	af, aok := keyFloat(a)
	bf, bok := keyFloat(b)
	if aok && bok {
		if af != bf {
			return af < bf
		}
		if ak != bk {
			return ak < bk
		}
		return numLess(a, b)
	}
	if ak != reflect.String || bk != reflect.String {
		return ak < bk
	}
	ar, br := a.String(), b.String()
	return ar < br
}
```